src/content/data-table-utils.js

```diff
@@ -64,7 +65,7 @@ export function sortDataTable<KeyColumnElementType>(
 
   function quickSort(left, right) {
     if (left < right) {
-      const pivot = right;
+      const pivot = (left + right) >> 1;
```

Comment:
The comment would be something like:

QuickSort's effectiveness depends on its ability to partition the sequence being sorted into two subsequences of roughly equal length: by halving the length at each recursion level, the subproblems get smaller faster. To get a good partition, we must choose a pivot value that is close to the median of the values in the sequence: by definition, half of the values will fall before the median, half after it.

*If the sequence is already mostly sorted*, then choosing the first or last value as the pivot is probably as far from the median as you could possibly get; this is a "pessimal" choice. Choosing the middle value as the pivot is likely to be much closer to the median.